dm/syncer/syncer.go

```diff
@@ -3173,8 +3176,12 @@ func (s *Syncer) Close() {
 
 	s.checkpoint.Close()
 
-	if err := s.schemaTracker.Close(); err != nil {
-		s.tctx.L().Error("fail to close schema tracker", log.ShortError(err))
+	// when close sycner without graceful, Close will be called before s.Process exit.
+	// if we close schematracker worker will pending on `s.checkpoint.Rollback(s.schemaTracker)` forever
+	if closeTracker {
+		if err := s.schemaTracker.Close(); err != nil {
```

Comment:
this may cause memory leak?

Reply:
sadly yes, but i can't find a better way to handle this, maybe don't rollback checkpoint when into `kill` branch?